chpl/chpl-service/src/main/java/gov/healthit/chpl/compliance/directreview/DirectReviewListingSharedStoreHandler.java

```diff
@@ -0,0 +1,120 @@
+package gov.healthit.chpl.compliance.directreview;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import gov.healthit.chpl.certifiedproduct.CertifiedProductDetailsManager;
+import gov.healthit.chpl.dao.DeveloperDAO;
+import gov.healthit.chpl.domain.CertifiedProductSearchDetails;
+import gov.healthit.chpl.domain.Developer;
+import gov.healthit.chpl.domain.compliance.DirectReview;
+import gov.healthit.chpl.domain.concept.CertificationEditionConcept;
+import gov.healthit.chpl.exception.EntityRetrievalException;
+import gov.healthit.chpl.search.ListingSearchService;
+import gov.healthit.chpl.search.domain.SearchRequest;
+import gov.healthit.chpl.sharedstore.listing.SharedListingStoreProvider;
+import gov.healthit.chpl.util.NullSafeEvaluator;
+import lombok.extern.log4j.Log4j2;
+import one.util.streamex.StreamEx;
+
+@Component
+@Log4j2
+public class DirectReviewListingSharedStoreHandler {
+    private SharedListingStoreProvider sharedListingStoreProvider;
+    private CertifiedProductDetailsManager certifiedProductDetailsManager;
+    private ListingSearchService listingSearchService;
+    private DeveloperDAO developerDAO;
+
+    @Autowired
+    public DirectReviewListingSharedStoreHandler(@Lazy SharedListingStoreProvider sharedListingStoreProvider,
+            @Lazy CertifiedProductDetailsManager certifiedProductDetailsManager, @Lazy ListingSearchService listingSearchService,
+            DeveloperDAO developerDAO) {
+        this.sharedListingStoreProvider = sharedListingStoreProvider;
+        this.certifiedProductDetailsManager = certifiedProductDetailsManager;
+        this.listingSearchService = listingSearchService;
+        this.developerDAO = developerDAO;
+    }
+
+    public void handler(List<DirectReview> allDirectReviews) {
+        getUniqueDevelopers(allDirectReviews).stream()
```

Comment:
This looks good - will do what we talked about. I was wondering if you changed the order of checks here if it would it be more efficient (enough to make it worthwhile)? Like, 99% of the time we expect the direct reviews won't have changed, so would it make sense to look for that first? Then the method becomes a no-op most of the time. And only if you find that any changed do you go through and get all the developers + all their listings. It's probably not that much extra work being done but since I have time I was thinking about that. I'm fine if you want to leave it how it is, too.